include/fuse_common.h

```diff
@@ -67,8 +75,16 @@ struct fuse_file_info {
 	/** Padding.  Do not use*/
 	unsigned int padding : 27;
 
-	/** File handle.  May be filled in by filesystem in open().
-	    Available in all other file operations */
+	/** File handle id.  May be filled in by filesystem in create,
+	 * open, and opendir().  Available in most other file operations on the
+	 * same file handle.
+	 *
+	 * Different file handles for the same file may have different file
+	 * handle IDs.  However, since an operation's file handle is not
+	 * guaranteed to be associated with the file descriptor ultimately
+	 * responsible for the operation, it is advised that file systems not
+	 * associated any credential-related information with the file handle.
```

Comment:
I don't like the second paragraph. The first sentence sound like a tautology, and the second sentence seems to imply that it would a good idea to put credentials into the file handle if e.g. direct_io is used. Can we just omit this paragraph?

Reply:
What about that sentence suggests that it would be a good idea to put credentials into the file handle?  I'm trying to suggest the opposite.

Reply:
You're saying it's not a good idea because an operation's file handle is not guaranteed to be associated with the file descriptor ultimately responsible for the operation. That is true, but it is neither the only reason nor the most important one.

Reply:
Well, "most important" is a subjective judgement.  What do you think is the most important reason?  I could add it.

Reply:
I don't want to call any one reason out at all, because this will just tempt people into neglecting the other reasons.